docs/source/ucs/uc2.md

```diff
@@ -0,0 +1,573 @@
+
+- [Folder] simulation_controller
+- [Folder] static
+- .gitmodules
+- app.py
+- deploy_heroku.sh
+- docker-compose.yml
+- Dockerfile
+- openAPI.yml
+- prepare_deployment.sh
+- requirements.txt
+
+## Create a singularity container
+
+In order to insure the correct working of the workflow, we create a container which install all the software we need
+
+Start from the image which already has mpi installed
+
+```docker
+FROM containers4hpc/base-mpich314:0.1.0
+SHELL [ "/bin/bash", "-c" ]
+```
+
+Be sure to install all the basic software you need, after which we can install OpenFOAM, this worflow work with OpenFOAM 6.
+
+```docker
+RUN sh -c "wget -O - http://dl.openfoam.org/gpg.key | apt-key add -" ;\
+	add-apt-repository http://dl.openfoam.org/ubuntu ;\
+	apt-get update ;\
+	apt-get install -y --no-install-recommends openfoam6 ;\
+	rm -rf /var/lib/apt/lists/* ;\
+	echo "source /opt/openfoam6/etc/bashrc" >> ~foam/.bashrc ;\
+	echo "export OMPI_MCA_btl_vader_single_copy_mechanism=none" >> ~foam/.bashrc
+```
+
+After which we can install RheoTools
+
+```docker
+RUN git clone https://github.com/fppimenta/rheoTool.git;\
+    cd rheoTool;\
+    git checkout 56c2701636e6eb59dc51799fd0636805be23510d
+
+RUN  source /opt/openfoam6/etc/bashrc;\
+    cd rheoTool/of60;\
+    ./downloadEigen;\
+	echo "export EIGEN_RHEO=$WM_PROJECT_USER_DIR/ThirdParty/Eigen3.2.9" >> ~foam/.bashrc
+```
+
+and install all the python dependencies that the workflow needs
+
+```python
+#install python packages
+RUN pip3 install numpy
+RUN pip3 install --upgrade pillow
+RUN pip3 install --upgrade matplotlib
+RUN pip3 install scipy
+RUN pip3 install vtk
+```
+
+Once the Dockerfile has all you need, you can create a docker image
+
+```
+ docker build -t uc2 .
+```
+
+after saving the image as a .tar file you can create a singularity container
+
+```
+singularity build (--sandbox) uc2 docker-archive://uc2.tar
+singularity build uc2.sif uc2
+```
+
+This container can be run anywhere
+
+## simulation controller
+
+Now we are working on the folder "simulation_controller" which contains several files that provide the function to create, start and stop a simulation as well as retrieving the simulation results.
+
+Let us have a look at the following files that are all found in the folder "simulation_controller"
+
+- \_\_init\_\_.py
+- config.py
+- paste_files_creation.py
+- post_processing.py
+- simulation_manager.py
+- simulation_base.py
+- simulation_hpc.py
+
+### \_\_init\_\_.py
+
+The file "\_\_init\_\_.py" is an empty file and its only purpose is that python allows to include all function via the regular
+package syntax as libraries. This means in all files, we can include classes and function of other files with simple commands. Usually, python has no problem to import a whole directory. But when it comes to importing a class from a file in a directory, this will raise an exception. But having such a "\_\_init\_\_.py" file allows to use the following notation
```

Comment:
```suggestion
package syntax as libraries. This means in all files, we can include classes and function of other files with simple commands. Usually, python has no problem importing a whole directory. But when it comes to importing a class from a file in a directory, this will raise an exception. But having such a "\_\_init\_\_.py" file allows to use the following notation
```